src/nvim/os/pty_process_unix.c

```diff
@@ -73,6 +73,12 @@ int pty_process_spawn(PtyProcess *ptyproc)
     goto error;
   }
 
+  // Other jobs and providers shouldn't get an copy of this file descriptor.
+  if (os_set_cloexec(master) == -1) {
```

Comment:
My comment had a typo, but I think we should be checking for `== -1`.

To clarify what I meant and correct the typo:

Good point that the two functions check for different return values.

The POSIX spec says that on succesfull *completion*, the return value of the call to `fcntl()` will be any value other than `-1`.

i.e. it could be `-2`.